src/data_hub/_api.py

```diff
@@ -132,6 +135,7 @@ def dynamic_function(
                 tokens["access"] = auth.refresh_tokens(tokens["refresh"])
                 auth.cache_tokens(tokens)
                 client.token = tokens["access"]
+                method = _get_method(client, method_path)
```

Comment:
Thought: Is this necessary? Does `_get_method` directly use `client.token`?

Reply:
i **think** it's needed (not entirely sure). 

in any case, this is simply a wrapper around `getattr` so i don't think it's doing anything __bad__ or slow. 

i think it's needed because the new method belongs to the client object with the correct token (i think???) i may have to trace the logic step by step